### PR TITLE
Add legacy release tests to docs repo

### PR DESCRIPTION
**Pull request by lwr20**
## Description

Adds post-release checks to legacy-releases in a similar way to what we have for newer branch-per-release releases.

## Release Note

```release-note
None required
```

### CODE REVIEW

File: release-scripts/tests/test_projectcalico_docs_updated.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+
+def test_latest_releases_redirects_correctly():
+    req = requests.get("https://docs.projectcalico.org/latest/release-notes")
+    assert req.status_code == 200
+
+    redirect = BeautifulSoup(req.content, features="html.parser").find("a", href=True)
+    assert (
+        redirect["href"]
+        == "https://docs.projectcalico.org/%s/release-notes/" % RELEASE_STREAM
```
Comment (lmm) [inline release-scripts/tests/test_projectcalico_docs_updated.py:29]: We can scrap this test I think. It's useful only for the latest version not older versions.

File: release-scripts/tests/test_tags_tars.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    # typha:
+    #     https://github.com/projectcalico/typha/releases/tag/v3.6.2
+    #     https://github.com/projectcalico/typha/archive/v3.6.2.zip
+    #     https://github.com/projectcalico/typha/archive/v3.6.2.tar.gz
+    # node: (as typha)
+    # kube-controllers: (as typha)
+    # dikastes: (as typha)
+    #
+    # networking-calico:
+    #     https://opendev.org/openstack/networking-calico/src/tag/3.6.0
```
Comment (lmm) [inline release-scripts/tests/test_tags_tars.py:169]: nit: comment is out of date

File: release-scripts/tests/test_projectcalico_docs_updated.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    req = requests.get("http://docs.projectcalico.org/latest")
+    assert req.status_code == 200
+
+
+def test_latest_redirects_correctly():
+    req = requests.get("https://docs.projectcalico.org/latest")
+    assert req.status_code == 200
+
+    redirect = BeautifulSoup(req.content, features="html.parser").find("a", href=True)
+    assert redirect["href"] == "https://docs.projectcalico.org/%s/" % RELEASE_STREAM
```
Comment (lmm) [inline release-scripts/tests/test_projectcalico_docs_updated.py:19]: This could also be removed I think. The test as-is fails

**Comment by caseydavenport** [comment]
@lwr20 @lmm this PR is quite old now - do we still want it?

**Comment by lwr20** [comment]
What releases are regarded as legacy?  If we think we'll ever do a legacy patch release, then it would be worth fixing and merging this.  Otherwise, probably not needed.

**Comment by lmm** [comment]
> What releases are regarded as legacy? 

I checked and it' v3.11 and older (this was when we had a version per folder)

**Pull request closed without merging**